src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,179 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"jenkinsci.org/plugins/durabletask/common"
+)
+
+type Shell int
+
+const (
+	CMD Shell = iota
+	POWERSHELL
+	PWSH
+)
+
+func (shell Shell) String() string {
+	switch shell {
+	case CMD:
+		return "cmd"
+	case POWERSHELL:
+		return "powershell"
+	case PWSH:
+		return "pwsh"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// Launches the script in a new session and waits for its completion.
+func launcher(wg *sync.WaitGroup, exitChan chan bool, shell string,
+	scriptPath string, resultPath string, outputPath string,
+	launchLogger *log.Logger, scriptLogger *log.Logger) {
+
+	defer wg.Done()
+	defer common.SignalFinished(exitChan)
+
+	if _, err := os.Stat(scriptPath); err != nil {
+		if os.IsNotExist(err) {
+			launchLogger.Printf("%s does not exist", scriptPath)
+			return
+		}
+	}
+
+	var scriptCmd *exec.Cmd
+	switch shell {
+	case CMD.String():
+		scriptCmd = exec.Command("cmd.exe", "/C", scriptPath)
+	case POWERSHELL.String():
+		shellCommand := fmt.Sprintf("[Console]::OutputEncoding = [Text.Encoding]::UTF8; .\\%s", scriptPath)
+		scriptCmd = exec.Command("powershell.exe", "-NoProfile", "-NonInteractive", "-ExecutionPolicy", "Bypass", "-Command", shellCommand)
+	case PWSH.String():
+	default:
+		launchLogger.Println("shell type not supported")
+		return
```

Comment:
should we flag the error and somehow `Exit` with an errorcode here?

Reply:
resolved in 8b29388f033068b9a614df700aaea11881d72f05